core/arch/riscv/kernel/mentry.S

```diff
@@ -0,0 +1,204 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#include <asm.S>
+#include <generated/asm-defines.h>
+#include <keep.h>
+#include <platform_config.h>
+#include <riscv.h>
+
+.section .data
+.balign 4
+
+#ifdef CFG_BOOT_SYNC_CPU
+.equ SEM_CPU_READY, 1
+#endif
+
+	/*
+	 * Setup sp to point to the top of the tmp stack for the current CPU:
+	 * sp is assigned:
+	 * stack_tmp + cpu_id * stack_tmp_stride - STACK_TMP_GUARD
+	 */
+.macro set_sp
+	/* Unsupported CPU, park it before it breaks something */
+	li	t1, CFG_TEE_CORE_NB_CORE
+	csrr	t0, CSR_MHARTID
+	bge	t0, t1, unhandled_cpu
+	la	t2, stack_tmp
+	lw	t1, stack_tmp_stride
+	/*
+	 * t0 is core pos + 1
+	 * t1 is value of stack_tmp_stride
+	 * t2 is value of stack_tmp
+	 */
+	mv	sp, t2
+	/* Don't use mul instruction because not all harts support it. */
```

Comment:
Could the multiplication be implemented using a macro instead, at least if there's a chance we'll need multiplications in assembly at other places too?